### DISCUSSION TITLE
Interview Status - COMPLETE 분리 논의

### DISCUSSION THREAD

**Discussion by HyeonbinSa** [woowacourse-teams/2022-ternoko#230] [Ideas]
## 지난 회의를 통해 정해진 사항
- 코치가 코멘트 작성 시 : COACH_COMPLETED
- 크루가 코멘트 작성 시 : CREW_COMPLETED

## 코드를 구현하면 로직 구상 시 발견 사항
현재 상황 예상
- 크루의 입장 : Interview의  Status가 COMMENT이거나 COACH_COMPLETED일 경우 [코멘트] 버튼 보여줌
- 코치의 입장 : Interview의  Status가 COMMENT이거나 CREW_COMPLETED일 경우 [코멘트] 버튼 보여줌
### Case 1. 크루
1. 크루가 코멘트 작성함. -> Status : CREW_COMPLETED
2. 코치는 코멘트를 작성하지 않았기 때문에 작성할 수 있음,  코치가 코멘트 작성함  ->  Status :  COACH_COMPLETED
3. Status만으로 비교한다면 프론트 엔드 측에서 크루에게 [피드백] 버튼을 보여줄 수 있음.
4. 반복
### Case 2. 코치
1. 코치가 코멘트 작성함. -> Status : COACH_COMPLETED
2. 크루는 코멘트를 작성하지 않았기 때문에 작성할 수 있음,  크루가 코멘트 작성함  ->  Status :  CREW_COMPLETED
3. Status만으로 비교한다면 프론트 엔드 측에서 코치에게 [피드백] 버튼을 보여줄 수 있음.
4. 반복

현재 Status 구조 상으로 위와 같은 문제가 발생할 수도 있겠다는 생각을 했습니다.
그래서 생각한 방식으로는 COMPLETE의 구성을 `CREW_COMPLETE`,`COACH_COMPLETE`,`COMPLETE`으로 분리하는 것이 좋겠다는 생각을 했습니다.
- `CREW_COMPLETE`  : 크루만 코멘트를 작성함.
- `COACH_COMPLETE`  : 코치만 코멘트를 작성함.
- `COMPLETE` : 둘 다 코멘트를 작성함.

제 의견은 여기까지이며 위와 같이 구현을 먼저하고 PR을 올리도록하겠습니다. 다른 분들의 의견에 따라 코드를 수정할 예정이며 자유롭게 의견 부탁드리겠습니다. 감사합니다.

**Comment by her0807** [comment]
제가 이해한 흐름을 설명해보겠습니다 .✨


## 크루 입장
크루입장과 코치입장에서 같은 문제가 생기는거 같아요. 
아래와 같이 이해했어요. 

1. 면담이 끝나면 `interivewStatus` -> `comment` 로 변경한다. 
2. 면담 후 자유로운 한마디 달면 `interviewStatus` -> `crewComplete` 로 변경
3. 코치도 자유로운 한마디 달면 `InterviewStatus` -> `complete` 로 변경 

여기서 프론트 입장을 생각해보면 `comment` 나 `coachCompete` 일 때 `진행중 면담 탭에서 면담을 보이게 하고,` 자유로운 한마디를 남길 수 있게 해둘 것 같아요. 

원래 `complete` 가 없을 때는 
> if( comment or coachCompete) 이면 자유로운 한마디를 남길 수 있게끔 했을테니까,  crewComplete 가 된 이후에 다시 coachCompete 가 되면 
크루 입장에서 또 자유로운 한마디를 남길 수 있게 되겠네요. 

이런 반복 상황을 만들지 않기 위해서 complete 를 추가 하신다는 말씀으로 이해했어요. 바니의 의견에 동의합니다.

**Comment by dongho108** [comment]
저도 의문을 가졌던 부분인데 바니가 제 생각과 동일하게 깔끔하게 정리해주셨네요!
